### PR TITLE
Update documentation to add v2 specific content

### PR DESCRIPTION
**Pull request by rohandubal**
## Description

This PR adds documentation pertaining to 2.0 related changes specifically. Will be merged post 2.0 release.

Kudos to @ChumpChief for authoring the container migration and recover docs and @Rick-Kirkham for reviewing them!

### CODE REVIEW

**Comment by pk-pranshu** [comment]
LGTM

**Review by pk-pranshu — APPROVED** [review]

File: docs/content/docs/data-structures/tree.md
Comment (Josmithr) [inline docs/content/docs/data-structures/tree.md:548]: I'm not sure I understand the new contents in the parenthesis. If the goal is to differentiate `nodeChanged`'s behavior from that of `treeChanged`, maybe just saying something like "i.e., when the values of any of the node's fields change. This does not include changes made to child nodes' fields." @alexvy86 you've been working in this space; what do you think?

Comment (Josmithr) [inline docs/content/docs/data-structures/tree.md:548]: Nit: I would recommend line-breaking Markdown contents along sentence boundaries. Makes future changes easier to review.
https://github.com/microsoft/FluidFramework/wiki/Markdown-Best-Practices#line-breaks-along-sentence-boundaries

Comment (alexvy86) [reply docs/content/docs/data-structures/tree.md:548]: Yeah, it's a finicky space for wording. I'd recommend this. Also replaces `eventType` with `eventName` to match the signature shown above.
```suggestion
`Tree.on` assigns the specified `listener` function to the specified `eventName` for the specified `node`.
The `node` can be any node of the tree.
The `eventName` can be either "treeChanged" or "nodeChanged".
`nodeChanged` fires whenever one or more properties of the specified node change.
`treeChanged` fires whenever one or more properties of the specified node or any node in its subtree, change.
We recommend looking at the documentation of each of the events for more details.
```

File: docs/content/docs/data-structures/tree.md
Comment (alexvy86) [inline docs/content/docs/data-structures/tree.md:550]: We should remove lines 550-552, `event.target` used to exist on a previous version of the events but does not exist on the current ones. We'll probably re-add it, but this document should not talk about it yet.

File: docs/content/docs/data-structures/tree.md
Comment (alexvy86) [inline docs/content/docs/data-structures/tree.md:557]:
```suggestion
const unsubscribe = Tree.on(myTreeNode, "nodeChanged", () => {...});

// Later at some point when the event subscription is not needed anymore
unsubscribe();
```

File: docs/content/docs/start/quick-start.md
Comment (alexvy86) [inline docs/content/docs/start/quick-start.md:17]: 2.0 feels better than 2, I think.
```suggestion
The demo app uses Fluid Framework 2.0.
```

File: docs/content/docs/start/tutorial.md
Comment (alexvy86) [inline docs/content/docs/start/tutorial.md:13]: Same as above.
```suggestion
The demo app uses Fluid Framework 2.0.
```

File: docs/content/docs/start/tutorial.md
Comment (alexvy86) [inline docs/content/docs/start/tutorial.md:63]: I'm a bit iffy with the wording here but don't know how I'd write it. I would think of each app as an independent thing, unless more than 1 app shares the same containers. If they are independent, I can have app 1 using its 1.x containers and that should not make me use compatibility "1" on a different app. But for any given app, if I expect to have _clients_ running FF 1.x, then that's probably when I want to use compatibility "1", until all clients are using FF 2.0 and the app has been updated as necessary to support it?
```suggestion
is compatible with. For new apps always use "2". Use "1" if you have apps running FluidFramework 1.x until you've migrated all the apps to version 2.
```

**Review by kashms — CHANGES_REQUESTED** [review]
Added some comments

File: docs/content/docs/start/quick-start.md
Comment (rohandubal) [reply docs/content/docs/start/quick-start.md:17]: Going with 2.X for consistency and future proofing.

**Comment by rohandubal** [comment]
Please hold off a review. Had a bad merge in last push

**Review by Josmithr — APPROVED** [review]
LGTM

**Review by kashms — APPROVED** [review]
🚢 it!

File: docs/content/docs/data-structures/tree.md
Comment (alexvy86) [reply docs/content/docs/data-structures/tree.md:556]: @rohandubal let's address this before merging.

File: docs/content/docs/start/tutorial.md
Comment (alexvy86) [inline docs/content/docs/start/tutorial.md:63]: I think @kashms suggestion [from this thread](https://github.com/microsoft/FluidFramework/pull/21586#discussion_r1651795974) still applies here:

> Yea, instead of "until you've migrated all the apps", I would suggest saying "app versions", "app builds" or simply "users". The idea is that you'll have a bunch of old app builds/users on old app builds and you would want to have a defined set of those app versions/users migrated over before flipping the switch.

**Pull request merged**
